Model/Adminhtml/Product/MassDelete.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Marsskom\ReservationAdmin\Model\Adminhtml\Product;
+
+use Magento\Framework\Data\Collection\AbstractDb;
+use Magento\Framework\Exception\LocalizedException;
+use Marsskom\ReservationAdmin\Api\Data\Product\MassAction\MassResultInterface;
+use Marsskom\ReservationAdmin\Api\Data\Product\MassAction\MassResultInterfaceFactory;
+use Marsskom\ReservationAdmin\Api\Data\Product\MassActionInterface;
+use Marsskom\ReservationAdmin\Api\Product\ReservationRepositoryInterface;
+use Psr\Log\LoggerInterface;
+
+class MassDelete implements MassActionInterface
+{
+    protected LoggerInterface $logger;
+
+    protected ReservationRepositoryInterface $reservRepo;
+
+    protected MassResultInterfaceFactory $massResultFactory;
+
+    /**
+     * Mass delete constructor.
+     *
+     * @param LoggerInterface                $logger
+     * @param ReservationRepositoryInterface $reservRepo
+     * @param MassResultInterfaceFactory     $massResultFactory
+     */
+    public function __construct(
+        LoggerInterface $logger,
+        ReservationRepositoryInterface $reservRepo,
+        MassResultInterfaceFactory $massResultFactory
+    ) {
+        $this->logger = $logger;
+        $this->reservRepo = $reservRepo;
+        $this->massResultFactory = $massResultFactory;
+    }
+
+    /**
+     * @inheridoc
+     */
+    public function process(AbstractDb $abstractDb): MassResultInterface
+    {
+        $deleted = 0;
+        $errored = 0;
+
+        foreach ($abstractDb->getItems() as $item) {
+            try {
+                $this->reservRepo->deleteById(
+                    (int) $item->getData('reservation_id')
+                );
```

Comment:
Mass action uses the repository for deleting the row?

Hmmm. Maybe, should create "really" mass action deletion?

Reply:
🥲 Yep, I should use `delete` instead of `deleteById`, because `$item` already is the instance of a reservation model.

However, I'll keep the repository usage here, because now the reservation is simplest table without many additional load and the deletion won't so heavy. At the same time usage of the repository allows us to write clear code with default Magento's interfaces.